sublime-package.json

```diff
@@ -508,7 +508,12 @@
                 "$ref": "sublime://settings/LSP#/definitions/ClientConfig"
               },
               "markdownDescription": "**DO NOT MODIFY THIS SETTING!** Use `\"clients\"` to override settings instead!",
-            }
+            },
+            "show_inlay_hints": {
+              "type": "boolean",
+              "default": true,
+              "markdownDescription": "Show inlay hints."
```

Comment:
If there's a toggle, as well as a json setting, what does the json setting do here? What if this setting is set to `false` and I run "Toggle Inlay Hints"?

Reply:
The json setting is there to preserve the user choice even after a ST restart.

The command is there to make it convenient for the user to toggle that json setting.

Reply:
> What if this setting is set to false and I run "Toggle Inlay Hints"?

If you put show_inlay_hints false and save the setting, 
the inlay hint phantoms in the view will go away.

If you run LSP: Toggle Inlay Hints, 
it will update that show_inlay_hints to true(because it was false),
And the inlay hint phantoms in the view will be shown.

Reply:
The reason why the toggle command was added, is because it was a suggested feature request 
See https://github.com/sublimelsp/LSP/pull/2018#issuecomment-1213077178

Reply:
Maybe then let's make in more generic:
```json
    {
        "command": "lsp_toggle_setting",
        "args": {"setting": "show_inlay_hints"},
        "keys": [
            "UNBOUND"
        ],
        "context": [
            {
                "key": "lsp.session_with_capability",
                "operator": "equal",
                "operand": "inlayHintProvider"
            }
        ]
    }
```

Then it can be reused for other features. The command would need to check that the given setting exists, and that it is a boolean setting. But it doesn't need to define the `capability` property, because this is already restricted by the keybinding context.

Reply:
Triggering this command rewrites the preferences file which then forces all running servers to restart. I really don't want this to work like that, especially when toggling inlay hints since it can take a while to restart and initialize a server.

While this command sounds useful for other purposes, toggling inlay hints should just result in enabling/disabling the inlay hint requests and updating inline hint phantoms, without restarting the server.

Reply:
I removed the toggle functionality in this PR, will be introduced in a future PR.